datalab/mlalpha/_dataset.py

```diff
@@ -11,407 +11,120 @@
-           on preprocessed data as described in feature_set.
-
+    self._schema = None
+    if schema is not None:
+      self._schema = schema
+    elif schema_file is not None:
+      with ml.util._file.open_local_or_gcs(schema_file, 'r') as f:
+        self._schema = json.load(f)
+    self._files = []
+    for file in files:
+      self._files += ml.util._file.glob_files(file)
+
+  @property
+  def files(self):
+    return self._files
+      
+  @property
+  def schema(self):
+    return self._schema
+  
+  def sample(self, n):
+    """ Samples data into a Pandas DataFrame.
     Args:
-      names: the names of the data to plot. Such as ['train']. If None, all data in the datasets
-          will be used.
-      columns: The list of names of columns to analyze. If None, all numeric columns
-          will be used.
-          If one column is provided, displays a scatter plot between the column and target
-          column, and a histogram of the column.
-          If two columns are provided, displays a scatter plot between them,
-          colored by target column.
-          If three columns are provided, displays a 3d scatter plot between them,
-          colored by target column.
-
-    Raises:
-      Exception if any column names are not found in the data or the columns are text.
-      Exception if columns are greater than 3 or less than 1.
+      n: number of sampled counts.
+    Returns:
+      A dataframe containing sampled data.
     """
-    self._load_to_dataframes()
-    if columns is None:
-      columns = [x for x in self._concatenated_data_frame 
-          if str(self._concatenated_data_frame[x].dtype) != 'object']
-    if len(columns) > 3 or len(columns) < 1:
-      raise 'Found %d columns. ' % len(columns) + \
-          'Use "columns" parameter to specify one, two or three columns.'
-    for column_name in columns:
-      if column_name not in self._concatenated_data_frame:
-        raise Exception('Cannot find column "%s"' % column_name)
-      if str(self._concatenated_data_frame[column_name].dtype) == 'object':
-        raise Exception('Cannot analyze text column "%s"' % column_name)
-
-    if names is None:
-      names = self._dataframes.keys()
-    if len(columns) == 1:
-      self._plot_x(names, columns[0])
-    elif len(columns) == 2:
-      self._plot_xy(names, columns[0], columns[1])
-    elif len(columns) == 3:
-      self._plot_xyz(names, columns[0], columns[1], columns[2])
-
-  def to_dataframes(self):
-    """Get the transformed data as a DataFrames
-
-    Returns: the transformed data in {name: dataframe} dictionary.
+    row_total_count = 0
+    row_counts = []
+    for file in self._files:
+      with ml.util._file.open_local_or_gcs(file, 'r') as f:
+        num_lines = sum(1 for line in f)
+        row_total_count += num_lines
+        row_counts.append(num_lines)
+
+    names = None
+    dtype = None
+    if self._schema:
```

Comment:
It can be None. I choose to make it optional in case the package knows the schema beforehand (like in inception).